src/coreclr/vm/runtimehandles.cpp

```diff
@@ -1143,6 +1143,41 @@ FCIMPL2(FC_BOOL_RET, RuntimeTypeHandle::CompareCanonicalHandles, ReflectClassBas
 }
 FCIMPLEND
 
+FCIMPL1(Object*, RuntimeTypeHandle::GetThreadLocalAllocInternal, MethodTable* pMT)
+{
+    FCALL_CONTRACT;
+
+    _ASSERTE(pMT != nullptr);
+    _ASSERTE(pMT->IsValueType());
+
+#ifdef FEATURE_64BIT_ALIGNMENT
+    if (pMT->RequiresAlign8())
+    {
+        return NULL;
+    }
+#endif
+
+    SIZE_T size = pMT->GetBaseSize();
+    _ASSERTE(size % DATA_ALIGNMENT == 0);
+
+    ee_alloc_context* allocContext = &t_runtime_thread_locals.alloc_context;
```

Comment:
Is the idea this is going handle `!UseThreadAllocationContexts` by returning null? We may need to add some comments or asserts to make it clear.

Reply:
I have added an explicit check to return false. I think single-proc case is rare enough to ignore this (useless) FCall overhead?